library/init/meta/interactive.lean

```diff
@@ -6,8 +6,11 @@ Authors: Leonardo de Moura
 prelude
 import init.meta.tactic init.meta.rewrite_tactic init.meta.simp_tactic
 import init.meta.smt.congruence_closure init.category.combinators
+import init.meta.lean_parser
 
 namespace interactive
+@[reducible] meta def parse {α : Type} (p : lean_parser α) : Type := α
```

Comment:
@gebner 
> @leodemoura Why do you think we'd get nasty error messages? These would be the same error messages the elaborator would return if you entered the term directly in Lean syntax.

I was referring to the weakly typed approach, where we use 
```lean
meta def parse {α : Type} (p : lean_parser pexpr) : Type := α
```
and making it clear that I'm aware of its limitations. In this setting, suppose we have a parameter `(a : parse T p)` where the action `p` is incorrect, that is, it produces a `pexpr` which will not be elaborated to a term of type `T`. I say the error messages will be nasty because even if the user provides something that should have type `T` it is getting an error message saying it doesn't. I'm acknowledge this is a limitation, but I think it is a small price to pay.